.lintr

```diff
@@ -1,5 +1,9 @@
 linters: linters_with_defaults(
     line_length_linter = line_length_linter(120),
     cyclocomp_linter = NULL,
-    object_usage_linter = NULL
+    object_usage_linter = NULL,
+    object_name_linter = object_name_linter(
+      styles = c("snake_case", "symbols"),
+      regexes = c(ANL = "^ANL_?[0-9A-Za-z_]*$", ADaM = "^r?AD[A-Z]{2,3}_?[0-9A-Za-z]*$")
```

Comment:
I'm not sure we should include lowercases on `ADXX_yyyy` variable names for the `ADSL_p` exception, but on the other hand, why not.

Regardless, please use `{2,5}` a for ADaM rule as it allows for sponsored CDISC datasets

```suggestion
      regexes = c(
        ANL = "^ANL_?[0-9A-Za-z_]*$",
        ADaM = "^r?AD[A-Z]{2,5}_?[0-9A-Za-z_]*$"
        )
```